src/thumbnailer_slope_optim.cc

```diff
@@ -26,7 +26,7 @@ Thumbnailer::Status Thumbnailer::GenerateAnimationSlopeOptim(
     CHECK_THUMBNAILER_STATUS(LossyEncodeNoSlopeOptim(webp_data));
     if (curr_anim_size == webp_data->size) break;
     curr_anim_size = webp_data->size;
-  } while (0);
+  } while (true);
```

Comment:
I think it would be safer to use a for loop so that we're sure we don't do too many iterations, and more importantly don't loop forever.
So something like:
const int kMaxIter = 5;
for (int i = 0; i < KMaxIter; ++i) {
  CHECK_THUMBNAILER_STATUS(LossyEncodeNoSlopeOptim(webp_data));
  if (curr_anim_size == webp_data->size) break;	    if (curr_anim_size == webp_data->size) break;
  curr_anim_size = webp_data->size;	    curr_anim_size = webp_data->size;
}

Reply:
Done!